cmd/pullrequest-init/main.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"context"
+	"flag"
+
+	"github.com/knative/pkg/logging"
+)
+
+const (
+	prFile = "pr.json"
+)
+
+var (
+	prURL = flag.String("url", "", "The url of the pull request to initialize.")
+	path  = flag.String("path", "", "Path of directory under which PR will be copied")
+	mode  = flag.String("mode", "download", "Whether to operate in download or upload mode")
+)
+
+// PullRequest represents a generic pull request resource.
+type PullRequest struct {
```

Comment:
this is pretty much a nit, but im curious why define these types in `main.go` when they aren't used in this file?